nucleo_f303re: Add lacking board-level dt-nodes description.

kuliga opened the pull request:
The nucleo_f303re board has a SoC equipped with i2c. Add lacking
devicetree and yaml description.

Signed-off-by: Jan Kuliga <jtkuliga@gmail.com>

kuliga committed:
boards: arm: nucleo_f303re: Add lacking board-level i2c dt description.

The nucleo_f303re board has a SoC equipped with i2c. Add lacking
devicetree and yaml description.

Signed-off-by: Jan Kuliga <jtkuliga@gmail.com>

erwango reviewed (changes requested):
Please update board doc

kuliga committed:
boards: arm: nucleo_f303re: doc: Fix board type typo.

The current board's documentation includes a board type typo.
Change the board type to the proper one.

Signed-off-by: Jan Kuliga <jtkuliga@gmail.com>

kuliga committed:
boards: arm: nucleo_f303re: Add lacking board-level spi dt description.

The nucleo_f303re board has a SoC equipped with spi. Add lacking
devicetree, doc and yaml description.

Signed-off-by: Jan Kuliga <jtkuliga@gmail.com>
